### PR TITLE
Remove exposed internal APIs

### LINKED ISSUE #652: Ensure there're no `@internal` APIs visible publicly
A `@public` exported API may expose a non-exported and/or non-public type - that is captured with our `extract-api` script. The output files show that some of the exposed APIs are `@internal`, which is wrong - only public APIs should be available publicly.

The `check-internal` script should be updated to capture `@internal` tags in the `extract-api` output files.

### PR DESCRIPTION
**Pull request by Yato333**
Closes #652
Unexported internal APIs that are used in exported ones (e.g. as parameters) will now have the release tag that matches the exported one.
APIs that were exposed for testing purposes were hidden using Contexts except for `FavoritePropertiesDataProvider` where there can be utilized a private member assignment hack.

### CODE REVIEW

File: scripts/checkInternal.js [hunk 1]
```diff
@@ -40,11 +40,9 @@ for (let lineIdx = 0; lineIdx < lines.length; ++lineIdx) {
     continue;
```
Comment (grigasp) [inline scripts/checkInternal.js:40]: Could we now change our regex to simply `@internal` (probably don't even need a regex for that)?

File: packages/components/src/test/propertygrid/UseUnifiedSelection.test.tsx [hunk 2]
```diff
@@ -4,17 +4,24 @@
...
+  PropertyDataProviderWithUnifiedSelectionProps,
+  usePropertyDataProviderWithUnifiedSelection,
+} from "../../presentation-components/propertygrid/UseUnifiedSelection";
 import { createTestECInstanceKey } from "../_helpers/Common";
-import { act, renderHook } from "../TestUtils";
+import { act } from "../TestUtils";
 
 describe("usePropertyDataProviderWithUnifiedSelection", () => {
   let selectionHandler: sinon.SinonStubbedInstance<SelectionHandler>;
+  let TestComponent: (props: Partial<PropertyDataProviderWithUnifiedSelectionProps>) => React.JSX.Element;
```
[the remark this answers is not in the document]
Comment (saskliutas) [reply packages/components/src/test/propertygrid/UseUnifiedSelection.test.tsx:24]: I aggree that `wrapper` prop should be used in this case. Similarly like https://github.com/iTwin/presentation/blob/5ae7122d66b1468235b6a8f6487a772a3d2f9da3/packages/hierarchies-react/src/test/internal/UseUnifiedSelection.test.tsx#L31 https://github.com/iTwin/presentation/blob/5ae7122d66b1468235b6a8f6487a772a3d2f9da3/packages/hierarchies-react/src/test/internal/UseUnifiedSelection.test.tsx#L87

File: packages/components/src/presentation-components/tree/DataProvider.ts [hunk 3]
```diff
@@ -279,7 +271,7 @@ export class PresentationTreeDataProvider implements IPresentationTreeDataProvid
       return createNodesAndCountResult(
         async () => this._dataSource.getNodesIterator(requestOptions),
         this.createBaseRequestOptions(),
-        (node, parentId) => this.createTreeNodeItem(node, parentId),
+        (node, parentId) => createTreeNodeItem(node, parentId, this.props),
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/src/presentation-components/tree/DataProvider.ts:274]: Why is this an issue? `PresentationTreeDataProviderProps` is compatible with `CreateTreeNodeItemProps`

[the remark this answers is not in the document]
Comment (grigasp) [reply packages/components/src/presentation-components/tree/DataProvider.ts:274]: Missed the fact that they're compatible. Agree that `CreateTreeNodeItemProps` should not be public.

File: packages/shared/src/shared/Values.ts [hunk 4]
```diff
@@ -44,7 +44,7 @@ export namespace InstanceKey {
 
 /**
  * A data structure for a 2d point.
- * @internal This is an internal type used in public API.
+ * @beta This is an internal type used in public API.
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/shared/src/shared/Values.ts]: Removed all similar comments

File: packages/unified-selection/src/unified-selection/types/IModel.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { Event } from "@itwin/presentation-shared";
 /**
  * Identifies the type of changes made to the `CoreIModelSelectionSet` to produce a `CoreSelectionSetEvent`.
  * @see https://www.itwinjs.org/reference/core-frontend/selectionset/selectionseteventtype/
- * @internal
+ * @internal This is an internal type used in public API.
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/unified-selection/src/unified-selection/types/IModel.ts]: Reverted

File: packages/unified-selection/src/unified-selection/CachingHiliteSetProvider.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ import { IMODEL_CLOSE_SELECTION_CLEAR_SOURCE, SelectionStorage } from "./Selecti
 
 /**
  * Props for creating a `CachingHiliteSetProvider` instance.
- * @internal Not exported through barrel, but used in public API as an argument. May be supplemented with optional attributes any time.
+ * @beta Not exported through barrel, but used in public API as an argument. May be supplemented with optional attributes any time.
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/unified-selection/src/unified-selection/CachingHiliteSetProvider.ts]: Removed all comments similar to this and added documentation for undocumented props

File: packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/LocalizationContext.tsx [hunk 7]
```diff
@@ -5,7 +5,7 @@
 
 import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
 
-/** @internal */
+/** @beta Not exported through barrel, but used in public API as an argument. */
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/LocalizationContext.tsx]: Removed all comments similar to this and added documentation for undocumented props

File: packages/components/src/presentation-components/favorite-properties/DataProvider.ts [hunk 8]
```diff
@@ -44,7 +41,13 @@ export interface FavoritePropertiesDataProviderProps {
  */
 export class FavoritePropertiesDataProvider implements IFavoritePropertiesDataProvider {
   private _customRuleset?: Ruleset | string;
-  private _propertyDataProviderFactory: (imodel: IModelConnection, ruleset?: Ruleset | string) => PresentationPropertyDataProvider;
+  // Stubbed in tests
+  /* istanbul ignore next */
+  private _propertyDataProviderFactory = (imodel: IModelConnection, ruleset?: Ruleset | string) => {
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/src/presentation-components/favorite-properties/DataProvider.ts]: Fixed

File: packages/components/src/presentation-components/common/SelectionHandlerContext.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+/** @internal */
+export function SelectionHandlerContextProvider({ selectionHandler, children }: PropsWithChildren<{ selectionHandler: SelectionHandler }>) {
+  return <SelectionHandlerContext.Provider value={selectionHandler}>{children}</SelectionHandlerContext.Provider>;
+}
+
+/** @internal */
+export function useSelectionHandlerContext() {
+  return useContext(SelectionHandlerContext);
+}
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/src/presentation-components/common/SelectionHandlerContext.tsx]: Fixed

File: packages/components/src/test/propertygrid/UseUnifiedSelection.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/src/test/propertygrid/UseUnifiedSelection.test.tsx]: Thanks, made suggested changes

File: packages/components/src/presentation-components/properties/editors/NavigationPropertyEditor.tsx [hunk 10]
```diff
@@ -13,7 +13,7 @@ import { navigationPropertyEditorContext } from "./NavigationPropertyEditorConte
  * Editor for navigation properties.
  *
  * **Note:** Should be used inside [[navigationPropertyEditorContext]].
- * @internal
+ * @beta Reexported elsewhere as deprecated beta.
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/src/presentation-components/properties/editors/NavigationPropertyEditor.tsx]: Made even little simpler by specifying a type for the exported constants.

File: scripts/checkInternal.js (hunk 1 above)
Comment (Yato333) [reply scripts/checkInternal.js]: Simplified

File: packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx [hunk 11]
```diff
@@ -19,14 +19,23 @@ type TreeNodeProps = ComponentPropsWithoutRef<typeof TreeNode>;
...
   getIcon?: (node: PresentationHierarchyNode) => ReactElement | undefined;
+  /** Returns a label for a given node. */
   getLabel?: (node: PresentationHierarchyNode) => ReactElement | undefined;
+  /** Returns sublabel for a given node. */
   getSublabel?: (node: PresentationHierarchyNode) => ReactElement | undefined;
+  /** Action to perform when the node is clicked. */
   onNodeClick?: (node: PresentationHierarchyNode, isSelected: boolean, event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+  /** Action to perform when a key is pressed when the node is hovered on. */
   onNodeKeyDown?: (node: PresentationHierarchyNode, isSelected: boolean, event: React.KeyboardEvent<HTMLElement>) => void;
+  /** Reloads the tree when an error occurs and `retry` button is clicked. */
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx:36]:
```suggestion
  /** A callback to reload a hierarchy level when an error occurs and `retry` button is clicked. */
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx [hunk 12]
```diff
@@ -19,17 +19,19 @@ type TreeNodeRendererProps = ComponentPropsWithoutRef<typeof TreeNodeRenderer>;
 
 /** @beta */
 interface TreeRendererOwnProps {
+  /** Root nodes of the tree. */
   rootNodes: PresentationTreeNode[];
+  /** Type of selection that is currently present. */
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx:24]:
```suggestion
  /** Active selection mode used by the tree. Defaults to `"single"`. */
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseSelectionHandler.ts [hunk 13]
```diff
@@ -18,7 +18,12 @@ import { useTree } from "./UseTree";
  */
 export type SelectionMode = "none" | "single" | "extended" | "multiple";
 
-/** @internal */
+/** Type of selection change.
+ * - `add` - a node was added to the selection.
+ * - `replace` - a selected node was replaced with a different one.
+ * - `remove` - a node was removed from the selection.
+ * @beta
+ */
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseSelectionHandler.ts:26]:
```suggestion
/**
 * Type of selection change.
 * - `add` - a node was added to the selection,
 * - `replace` - a selected node was replaced with a different one,
 * - `remove` - a node was removed from the selection.
 *
 * @beta
 */
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 14]
```diff
@@ -110,19 +118,28 @@ interface ReloadTreeCommonOptions {
   state?: "keep" | "discard" | "reset";
 }
 
-/** @beta */
+/**
+ * Options for the full tree reload.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:122]:
```suggestion
 * Options for full tree reload.
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 15]
```diff
@@ -110,19 +118,28 @@ interface ReloadTreeCommonOptions {
...
+ * @beta
+ */
 type FullTreeReloadOptions = {
   /** Specifies that data source changed and caches should be cleared before reloading tree. */
   dataSourceChanged?: true;
 } & ReloadTreeCommonOptions;
 
-/** @beta */
+/**
+ * Options for the subtree reload.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:131]:
```suggestion
 * Options for subtree reload.
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 16]
```diff
@@ -110,19 +118,28 @@ interface ReloadTreeCommonOptions {
...
+ * @beta
+ */
 type SubtreeReloadOptions = {
   /** Specifies parent node under which sub tree should be reloaded. */
   parentNodeId: string;
 } & ReloadTreeCommonOptions;
 
-/** @beta */
+/**
+ * Options for reloading a tree or a subtree.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:140]:
```suggestion
 * Options for doing either full or a sub tree reload.
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 17]
```diff
@@ -136,11 +153,25 @@ interface UseTreeResult {
    * or tree is reloading.
    */
   isLoading: boolean;
+  /**
+   * Action to perform when tree has to be reloaded.
+   */
   reloadTree: (options?: ReloadTreeOptions) => void;
+  /**
+   * Action to perform when a node is expanded or collapsed.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:161]:
```suggestion
   * A function that should be called to either expand or collapse the given node.
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 18]
```diff
@@ -136,11 +153,25 @@ interface UseTreeResult {
    * or tree is reloading.
    */
   isLoading: boolean;
+  /**
+   * Action to perform when tree has to be reloaded.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:157]:
```suggestion
   * A function that should be called to reload the tree.
```

File: packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts [hunk 19]
```diff
@@ -136,11 +153,25 @@ interface UseTreeResult {
...
+   */
   reloadTree: (options?: ReloadTreeOptions) => void;
+  /**
+   * Action to perform when a node is expanded or collapsed.
+   */
   expandNode: (nodeId: string, isExpanded: boolean) => void;
+  /**
+   * Action to perform when nodes are selected.
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/UseTree.ts:165]:
```suggestion
   * A function that should be called to select nodes in the tree.
```

File: packages/components/src/presentation-components/tree/DataProvider.ts [hunk 20]
```diff
@@ -127,26 +127,18 @@ export interface PresentationTreeDataProviderDataSourceEntryPoints {
  * @public
  */
 export class PresentationTreeDataProvider implements IPresentationTreeDataProvider, IDisposable {
-  private _imodel: IModelConnection;
-  private _ruleset: RulesetOrId;
   private _pagingSize?: number;
```
Comment (grigasp) [inline packages/components/src/presentation-components/tree/DataProvider.ts:130]: I think we should remove this in favor of paging size we have in props

Comment (Yato333) [reply packages/components/src/presentation-components/tree/DataProvider.ts]: Done

File: packages/components/api/presentation-components.api.md [hunk 21]
```diff
@@ -575,6 +548,8 @@ export class PresentationTreeDataProvider implements IPresentationTreeDataProvid
     get imodel(): IModelConnection;
     get pagingSize(): number | undefined;
     set pagingSize(value: number | undefined);
+    // (undocumented)
+    props: PresentationTreeDataProviderProps;
```
[the remark this answers is not in the document]
Comment (Yato333) [reply packages/components/api/presentation-components.api.md]: Fixed

**Review by grigasp — APPROVED** [review]

**Pull request merged**